### PR TITLE
refactor: adopt hybrid src architecture and align docs

### PR DESCRIPTION
**Pull request by naml14**
## Summary

This PR applies the hybrid src/ architecture refactor and aligns the documentation to the new structure.

### What changed
- Reorganized src/ into:
  - src/core/types.ts
  - src/reactivity/signals.ts
  - src/render/** (pipeline + engines + strategy)
  - src/features/**
  - root orchestration/public files remain (app.ts, router.ts, scheduler.ts, ssr.ts, testing.ts, index.ts)
- Updated internal imports across runtime, render pipeline, features, tests, and demo to match new paths.
- Added/updated architecture documentation and references:
  - docs/ARCHITECTURE.md
  - README.md architecture entry
  - .github/AGENTS.md, .github/CONTRIBUTING.md, .github/pull_request_template.md
  - route/plan docs aligned to current tree
- Documented architectural exception: src/render/commit.ts -> src/features/style.ts.

## Validation
- bun run typecheck
- bun test (suite green in this change set; known env-specific skips remain documented)
 
## Notes
- Public API remains stable via src/index.ts re-exports.
- This is a structural refactor (no intended functional behavior changes).

## Summary by Sourcery

Adopta una arquitectura híbrida basada en `src/` reorganizando el runtime central, el pipeline de renderizado y los módulos de funcionalidades en subcarpetas dedicadas, manteniendo la API pública estable mediante re-exportaciones actualizadas, y ampliando el sistema de portales con una bandera `cssManaged` que controla la responsabilidad del layout entre CSS y el framework.

New Features:
- Añadir opciones `cssManaged` a los portales para que los consumidores puedan elegir entre un layout gestionado por CSS y un posicionamiento en línea gestionado por el framework para los elementos hijos del portal.

Enhancements:
- Actualizar los internos del pipeline de renderizado (`prepare`, `reflow`, `diff`, `commit`) para respetar la bandera `cssManaged` de los portales, garantizando que los hijos de los portales puedan participar opcionalmente en el layout del framework, manteniendo el comportamiento por defecto sin cambios.
- Refinar las estructuras de operaciones sobre el DOM y de mapeo de portales para incluir metadatos de layout de portales sin alterar el comportamiento existente de los portales por defecto.

Documentation:
- Introducir documentación arquitectónica detallada para el nuevo layout híbrido bajo `src/` y actualizar los planes, guías y diagramas del README existentes para que hagan referencia a la nueva estructura de módulos y a las reglas de dependencias.

Tests:
- Ampliar las pruebas de portales, layout y commit para cubrir el comportamiento de portales con `cssManaged` a lo largo de las fases de `prepare`, `reflow` y `commit`, y actualizar las importaciones de tests a las nuevas ubicaciones de los módulos de render y core.

Chores:
- Mover los módulos de core, reactividad, render y funcionalidades a una estructura de carpetas híbrida bajo `src/` y ajustar todas las importaciones internas, demos y referencias de tooling en consecuencia, incluyendo la documentación de la dependencia intencional de `render/commit.ts` a `features/style.ts`.

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Adopt a hybrid src/ architecture by reorganizing core runtime, render pipeline, and feature modules into dedicated subfolders while keeping the public API stable via updated re-exports, and extend the portal system with a cssManaged flag that controls layout responsibility between CSS and the framework.

New Features:
- Add cssManaged options to portals so callers can choose between CSS-managed layout and framework-managed inline positioning for portal children.

Enhancements:
- Update render pipeline internals (prepare, reflow, diff, commit) to respect the portal cssManaged flag, ensuring portal children can optionally participate in framework layout while keeping default behavior unchanged.
- Refine DOM operation and portal mapping structures to carry portal layout metadata without altering existing behavior for default portals.

Documentation:
- Introduce detailed architectural documentation for the new hybrid src/ layout and update existing plans, guides, and README diagrams to reference the new module structure and dependency rules.

Tests:
- Expand portal, layout, and commit tests to cover cssManaged portal behavior across prepare, reflow, and commit phases, and update test imports to the new render and core module locations.

Chores:
- Move core, reactivity, render, and feature modules into a hybrid folder structure under src/ and adjust all internal imports, demos, and tooling references accordingly, including documenting the intentional render/commit.ts to features/style.ts dependency.

</details>

### CODE REVIEW

File: tests/commit.test.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { describe, test, expect, beforeAll } from 'bun:test'
 import { Window } from 'happy-dom'
-import { applyOps, commitFull, type DOMOperation } from '../src/commit.js'
-import { defineComponent } from '../src/component.js'
-import { prepare, resetIndexCounter } from '../src/prepare.js'
-import { reflow, createLayoutResult } from '../src/reflow.js'
+import { applyOps, commitFull, type DOMOperation } from '../src/render/commit.js'
+import { defineComponent } from '../src/render/component.js'
+import { prepare, resetIndexCounter } from '../src/render/prepare.js'
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/commit.test.ts]: ✅ Resuelto en `17b8ca6`: eliminé `resetIndexCounter` del import en `tests/commit.test.ts` porque no se usa.

File: tests/commit.test.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+import { reflow, createLayoutResult } from '../src/render/reflow.js'
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/commit.test.ts]: ✅ Resuelto en `17b8ca6`: eliminé `createLayoutResult` del import en `tests/commit.test.ts` (import no utilizado).

File: tests/edge-cases.test.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import { describe, test, expect, beforeAll } from 'bun:test'
 import { Window } from 'happy-dom'
-import { defineComponent } from '../src/component.js'
-import { prepare, resetIndexCounter, countNodes } from '../src/prepare.js'
-import { reflow } from '../src/reflow.js'
-import { commitFull } from '../src/commit.js'
-import type { DOMState } from '../src/commit.js'
-import type { LayoutConstraints } from '../src/types.js'
+import { defineComponent } from '../src/render/component.js'
+import { prepare, resetIndexCounter, countNodes } from '../src/render/prepare.js'
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/edge-cases.test.ts]: ✅ Resuelto en `17b8ca6`: eliminé `resetIndexCounter` del import en `tests/edge-cases.test.ts` (no utilizado).

File: tests/reflow.test.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { describe, test, expect, mock, beforeEach } from 'bun:test'
-import { defineComponent } from '../src/component.js'
-import { prepare, resetIndexCounter } from '../src/prepare.js'
-import { reflow, createLayoutResult } from '../src/reflow.js'
-import type { PreparedComponent } from '../src/types.js'
+import { defineComponent } from '../src/render/component.js'
+import { prepare, resetIndexCounter } from '../src/render/prepare.js'
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/reflow.test.ts]: ✅ Resuelto en `17b8ca6`: eliminé `resetIndexCounter` del import en `tests/reflow.test.ts` (no utilizado).

File: tests/portal.test.ts [hunk 5]
```diff
@@ -835,3 +835,212 @@ describe('portal reactivity — signal update inside portal', () => {
...
+describe('reflow — cssManaged:false portal children participate in layout (issue #9)', () => {
+  let target: HTMLElement
+
+  beforeEach(() => {
+    resetIndexCounter()
+    target = { tagName: 'DIV' } as unknown as HTMLElement
+  })
+
+  test('cssManaged:false portal slot is still 0x0 in parent', () => {
+    const component = defineComponent(() => ({
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/portal.test.ts:930]: ✅ Resuelto en `17b8ca6`: añadí cobertura de portales anidados con flags mixtos `cssManaged` en `tests/portal.test.ts` (caso outer true/inner false y outer false/inner true), verificando comportamiento esperado por subárbol.

File: .github/AGENTS.md [hunk 6]
```diff
@@ -182,7 +193,7 @@ Todas las posiciones (`x`, `y`) son **relativas al padre directo**, no absolutas
 - **Columna:** height = suma de heights de hijos + gaps
 - **Fila:** height = máximo de heights de hijos
 - **Padre sin height explícito:** se calcula automáticamente desde los hijos
-- **`measureFlex`** calcula el height del padre al final — `reflow.ts` NO debe sobreescribirlo
+- **`measureFlex`** calcula el height del padre al final — `render/reflow.ts` NO debe sobreescribirlo
```
[the remark this answers is not in the document]
Comment (naml14) [reply .github/AGENTS.md]: ✅ Resuelto en `17b8ca6`: corregí la redacción en `.github/AGENTS.md` a “sobrescribirlo”.

File: src/render/reflow.ts [hunk 7]
```diff
@@ -46,10 +47,39 @@ export function reflow(
...
+        layoutNode(child, constraints, result, lineHeight)
+      }
+    }
+    // Don't recurse deeper — layoutNode above will handle subtrees of each child
+    return
+  }
+  for (const child of getPreparedChildren(node)) {
+    reflowPortalChildren(child, constraints, result, lineHeight)
+  }
+}
```
[the remark this answers is not in the document]
Comment (naml14) [reply src/render/reflow.ts:81]: ℹ️ Revisado en detalle: esta observación NO aplica como bug en la implementación actual. Probé remover la segunda pasada y rompía casos válidos de `cssManaged:false` (fallaban tests de portal por layout no poblado cuando el portal es omitido por `measureSimple/measureFlex`). Por eso mantengo `reflowPortalChildren` por ahora; su eliminación requeriría rediseñar el flujo de layout del hot path para no perder cobertura funcional.

File: tests/commit.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (naml14) [reply tests/commit.test.ts]: ✅ Resuelto en 17b8ca6: eliminé resetIndexCounter del import en tests/commit.test.ts porque no se usa.

File: tests/commit.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (naml14) [reply tests/commit.test.ts]: ✅ Resuelto en 17b8ca6: eliminé createLayoutResult del import en tests/commit.test.ts (import no utilizado).

File: tests/edge-cases.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (naml14) [reply tests/edge-cases.test.ts]: ✅ Resuelto en 17b8ca6: eliminé resetIndexCounter del import en tests/edge-cases.test.ts (no utilizado).

File: tests/reflow.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (naml14) [reply tests/reflow.test.ts]: ✅ Resuelto en 17b8ca6: eliminé resetIndexCounter del import en tests/reflow.test.ts (no utilizado).

File: tests/portal.test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (naml14) [reply tests/portal.test.ts:930]: ✅ Resuelto en 17b8ca6: añadí cobertura de portales anidados con flags mixtos cssManaged en tests/portal.test.ts (outer true/inner false y outer false/inner true), verificando el comportamiento por subárbol.

File: .github/AGENTS.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (naml14) [reply .github/AGENTS.md]: ✅ Resuelto en 17b8ca6: corregí la redacción en .github/AGENTS.md a "sobrescribirlo".

File: src/render/reflow.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (naml14) [reply src/render/reflow.ts:81]: ℹ️ Revisado en detalle: esta observación NO aplica como bug en la implementación actual. Al remover la segunda pasada fallaban casos válidos de cssManaged:false (tests de portal), porque measureSimple/measureFlex omiten portales y no poblan layout de hijos sin reflowPortalChildren. Mantengo esa segunda pasada por corrección funcional; eliminarla requiere rediseño del flujo de layout.

**Comment by naml14** [comment]
Nota sobre el punto repetido (Gemini + Sourcery) de posible redundancia en 
eflowPortalChildren: lo validé con pruebas y al quitar esa segunda pasada se rompían casos válidos de cssManaged:false (los motores measureSimple/measureFlex omiten portales y no rellenan layout de esos hijos sin esa pasada). Por eso se mantiene por corrección funcional en esta iteración; cualquier simplificación futura requiere rediseñar el flujo para no perder cobertura.

File: docs/ARCHITECTURE.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+| `reactivity/` | `core/` |
+| `render/` | `core/`, `reactivity/` |
+| `features/` | `core/`, `reactivity/`, `render/` (public contracts only) |
+| Root files | anywhere as needed |
+
+## Architectural Exception
+
+**`render/commit.ts` → `features/style.ts`**: Allowed for CSS application during the commit phase.  
+Decoupling requires functional refactoring (strategy pattern or callback injection) which is out of scope for the structural change.  
+Documented in [PLAN-REFACTOR-SRC-HIBRIDO.md](./PLAN-REFACTOR-SRC-HIBRIDO.md) and marked with an inline comment in `src/render/commit.ts`.
```
[the remark this answers is not in the document]
Comment (naml14) [reply docs/ARCHITECTURE.md]: ✅ Resuelto en `05f5f4c`: actualicé `docs/ARCHITECTURE.md` para aclarar que las reglas aplican a imports de runtime y que existen type-only bridges actuales (`core/types.ts` y `render/{prepare,diff}.ts` hacia `features/style.ts`). Con esto la documentación queda alineada con el estado real del código.
